### PR TITLE
Convert feature examples to es6

### PR DESCRIPTION
**Pull request by alexdavid**
@kevgo @charlierudolph This converts all of the coffeescript in the Gherkin files to es6 javascript

### CODE REVIEW

File: src/base_controller.coffee [hunk 1]
```diff
@@ -14,8 +14,9 @@ class BaseController
   getActions: ({only, except} = {}) ->
     only = [].concat only ? []
     except = [].concat except ? []
-    for action of this
-      continue if BaseController::[action]?
+    for action in Object.getOwnPropertyNames @constructor.prototype
```
Comment (charlierudolph) [inline src/base_controller.coffee:17]: Just curious what made you choose `Object.getOwnPropertyNames` over `Object.keys`

File: test/feature/use_middleware_helper.feature [hunk 2]
```diff
@@ -2,31 +2,44 @@ Feature: BaseController useMiddleware helper
...
-        GET '/users', to: 'users#index'
-        GET '/users/:id', to: 'users#show'
+      module.exports = ({ GET }) => {
+        GET('/users', { to: 'users#index' });
+        GET('/users/:id', { to: 'users#show' });
+      }
+      """
+    And a file "middleware.js" with the content
+      """
+        exports.fooInjector = function(req, res, next) {
```
Comment (charlierudolph) [inline test/feature/use_middleware_helper.feature]: Any reason these lines are indented more than the start of the quotes?

File: test/feature/use_middleware_helper.feature [hunk 3]
```diff
@@ -39,88 +52,78 @@ Feature: BaseController useMiddleware helper
...
     And an exprestive app using defaults
     When making a <REQUEST> request to "<URL>"
     Then the response body should be "<RESPONSE BODY>"
 
     Examples:
       | REQUEST | URL      | RESPONSE BODY |
-      | GET     | /users   | foo           |
-      | GET     | /users/1 | No Custom     |
+      | GET     | /users   | object        |
+      | GET     | /users/1 | undefined     |
```
Comment (charlierudolph) [inline test/feature/use_middleware_helper.feature:103]: I think I liked the old one better instead of now returning typeof.

**Comment by charlierudolph** [comment]
Looks good

File: src/base_controller.coffee (hunk 1 above)
Comment (alexdavid) [inline src/base_controller.coffee:17]: For some reason `Object.keys` doesn't return the keys of ES6 classes. I'm not sure what magic they are doing but `Object.getOwnPropertyNames` seems to work

File: test/feature/use_middleware_helper.feature (hunk 2 above)
Comment (alexdavid) [inline test/feature/use_middleware_helper.feature]: Oops my mistake

File: test/feature/use_middleware_helper.feature [hunk 4]
```diff
@@ -39,54 +52,46 @@ Feature: BaseController useMiddleware helper
...
+      middleware = require('../middleware');
+
+      module.exports = class UsersController extends BaseController {
+        constructor() {
+          super();
+          this.useMiddleware(middleware.fooInjector, { only: 'index' });
+        }
+
+        index (req, res) { res.end(String(req.custom)); }
+        show  (req, res) { res.end(String(req.custom)); }
```
Comment (alexdavid) [inline test/feature/use_middleware_helper.feature:93]: @charlierudolph what do you think about this instead of `typeof`?

Comment (charlierudolph) [inline test/feature/use_middleware_helper.feature:93]: :+1:

**Pull request closed without merging**